hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,294 @@
+            delete interfaces {port} native-vlan-id
+            """.format(port=port)
+        self._set_commit_config(remove_all_vlans)
+
+    def _get_port_networks(self, port):
+        """ Returns list of all the vlans shared with the port. """
+        port_info = self._interface_info(port)
+        return port_info[port]['vlans'] 
+
+    def _set_port_state(self, port, state):
+        """ Disables or enables the `port`. 
+        Depending on the value of `state`.
+        """
+
+        if state == "enable":
+            set_port_state = """
+            delete interface {port} disable
+            """.format(port=port)
+        elif state == "disable":
+            set_port_state = """
+            set interface {port} disable
+            """.format(port=port)
+        self._set_commit_config(remove_vlan)
+
+    def revert_port(self, port):
+        """Resets the port to the factory default.
+        Running this function will replace the port settings
+        with the ones stored in the jinja template `jinja_templates/`
+        For a physical port, it will use the template `revert_nonchannelized_port.j2`
+        For a channelized port ( 4 10G ports made from 1 40G port) it will use
+        template `revert_channelized_port.j2`
+        """
+        port_name = { 'port_name': port }
+        jun = self.session()
+        rm_curr_port_config = """
+        delete interfaces {iface}
+        """.format(iface=port_name['port_name'])
+
+        if port[-2] == ":":
+            base_config='junos/jinja_templates/revert_channelized_port.j2'
+        else:
+            base_config='junos/jinja_templates/revert_nonchannelized_port.j2'
+        
+        try:
+            with jun:
+                jun.cfg.load(rm_curr_port_config, format="set")
+                jun.cfg.load( 
+                        template_path = base_config, template_vars=port_name,
+                        format='text'
+                        )
+
+                if jun.cfg.commit_check:
+                    jun.cfg.commit
+                else:
+                    jun.cfg.rollback
+                
+        except ConnectError as err:
+            print ("Cannot connect to device: {0}".format(err))
+            sys.exit(1)
+        except Exception as err:
+            print (err)
+
+    def modify_port(self, port, channel, network_id):
+        """ Changes vlan assignment to the port.
+        """
+
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if network_id is None:
+                self._remove_native_vlan(interface)
+            else:
+                self._set_native_vlan(interface, network_id)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                    " switch!"
+             #Discuss this with Kristi
```

Comment:
What's this comment for? can you please elaborate?

Reply:
it was a `note to myself`. removed it.